gems/aws-sdk-core/lib/aws-sdk-core/client_stubs.rb

```diff
@@ -179,6 +178,16 @@ def stub_responses(operation_name, *stubs)
       end
     end
 
+    def api_requests
+      if config.stub_responses
+        @api_requests
+      else
+        msg = 'you must enable stubbing in order to client requests;'
```

Comment:
Maybe just say:

"This method is only implemented for stubbed clients, and is available when you enable stubbing in the constructor with `stub_responses: true`."

I think this wording is important because it can read like you're supposed to turn on stubbing, when really you're just using this if you know you need it, like for testing.

I'd also raise a typed exception, perhaps `NotImplementedError`.

Reply:
The `stub_responses` method raises a generic `RuntimeError`, should I leave the error handling the way it is for consistency or use a typed exception?